Source/WebCore/dom/Node.cpp

```diff
@@ -621,17 +635,23 @@ ExceptionOr<RefPtr<Node>> Node::convertNodesOrStringsIntoNode(FixedVector<NodeOr
 }
 
 // https://dom.spec.whatwg.org/#converting-nodes-into-a-node except this returns a NodeVector
-ExceptionOr<NodeVector> Node::convertNodesOrStringsIntoNodeVector(FixedVector<NodeOrString>&& nodeOrStringVector)
+ExceptionOr<NodeVector> Node::convertNodesOrStringsOrTrustedScriptsIntoNodeVector(RefPtr<Node> parent, FixedVector<NodeOrStringOrTrustedScript>&& vector)
 {
-    if (nodeOrStringVector.isEmpty())
+    if (vector.isEmpty())
         return NodeVector { };
 
     Ref document = this->document();
     NodeVector nodeVector;
-    nodeVector.reserveInitialCapacity(nodeOrStringVector.size());
-    for (auto& variant : nodeOrStringVector) {
-        if (std::holds_alternative<String>(variant)) {
-            nodeVector.append(Text::create(document, WTFMove(std::get<String>(variant))));
+    nodeVector.reserveInitialCapacity(vector.size());
```

Comment:
I think this is not properly doing step 2 before doing the rest of the algorithm:
https://whatpr.org/dom/1258.html#converting-nodes-into-a-node

Also should there not be a feature flag for the case TT is not enabled somewhere here?

Reply:
The feature flags are handled in the TT helper functions.

Reply:
Where do the feature flags come into play? It seems with this change String is doing an extra check for parent being script, and Node is doing an extra check if it is a TrustedScript, despite the flag being turned off?

Reply:
Okay yeah extra code is ran in the situation where the flag is off. But there's no behaviour change, the flag is in trustedTypeCompliantString so that function call will just return the string unchanged.

I guess I should add a feature flag check in higher up to skip those checks?

Reply:
Right, I would aim to avoid any speed regressions when TT is tuned off.

Reply:
Added flag checks higher up the call stack